include/kernel.h

```diff
@@ -3119,7 +3119,7 @@ typedef int (*k_alert_handler_t)(struct k_alert *alert);
  */
 
 #define K_ALERT_DEFAULT NULL
-#define K_ALERT_IGNORE ((void *)(-1))
+#define K_ALERT_IGNORE ((k_alert_handler_t)0xFFFFFFFF)
```

Comment:
Don't understand this one, what's the violation?  There's a typecast either way.

Reply:
(void *) typecasting should only be done for 0 i.e **_(void *)0_**
Any thing other than this is a violation of misra.
I know there are a lot of such places in zephyr. Some of them cant be fixed. But this current implementation is quite straight forward and using the correct typecast fixes the violation.